spec.html

```diff
@@ -120,36 +138,94 @@ <h1>Array.prototype.spliced ( _start_, _deleteCount_, ..._items_ )</h1>
+                        1. Perform ? CreateDataPropertyOrThrow(_A_, _Pk_, _kValue_).
+                        1. Set _k_ to _k_ + 1.
+                    1. Let _k_ be _actualStart_.
+                    1. For each element _E_ of _items_, do
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Perform ? CreateDataPropertyOrThrow(_A_, _Pk_, _E_).
+                        1. Set _k_ to _k_ + 1.
+                    1. Repeat, while _k_ &lt; _newLen_,
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Let _from_ be ! ToString(𝔽(_k_ + _actualDeleteCount_ - _insertCount_)).
+                        1. Let _fromValue_ be ? Get(_O_, _from_).
+                        1. Perform ? CreateDataPropertyOrThrow(_A_, _Pk_, _fromValue_).
+                        1. Set _k_ to _k_ + 1.
                     1. Return _A_.
                 </emu-alg>
             </emu-clause>
 
             <emu-clause id="sec-array.prototype.unshifted">
-                <h1>Array.prototype.unshifted ( ..._args_ )</h1>
+                <h1>Array.prototype.unshifted ( ..._items_ )</h1>
 
                 <p>When the *unshifted* method is called, the following steps are taken:</p>
 
                 <emu-alg>
-                    1. Let _O_ be ? ToObject(*this* value).
-                    1. Let _A_ be ArrayClone(_O_).
-                    1. Perform ? Call(Array.prototype.unshift, _A_, «..._args_»).
+                    1. Let _itemCount_ be the number of elements in _items_.
+                    1. Let _O_ be the *this* value.
+                    1. Let _len_ be ? LengthOfArrayLike(_O_).
+                    1. Let _newLen_ be _len_ + _itemCount_.
+                    1. Let _A_ be ? ArraySpeciesCreate(_O_, 𝔽(_newLen_)).
+                    1. Let _k_ be 0.
+                    1. For each element _E_ of _items_, do
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Perform ? CreateDataPropertyOrThrow(_A_, _Pk_, _E_).
+                        1. Set _k_ to _k_ + 1.
+                    1. Let _from_ be 0.
+                    1. Repeat, while _k_ &lt; _newLen_,
+                        1. Let _Pk_ be ! ToString(𝔽(_k_)).
+                        1. Let _Pfrom_ be ! ToString(𝔽(_from_)).
+                        1. Let _fromValue_ be ? Get(_O_, _Pfrom_).
+                        1. Perform ? CreateDataPropertyOrThrow(_A_, _Pk_, _fromValue_).
+                        1. Set _k_ to _k_ + 1.
+                        1. Set _from_ to _from_ + 1.
                     1. Return _A_.
                 </emu-alg>
             </emu-clause>
 
             <emu-clause id="sec-array.prototype.with">
                 <h1>Array.prototype.with ( _index_, _value_ )</h1>
 
-                <p>When the *unshifted* method is called, the following steps are taken:</p>
+                <p>When the *with* method is called, the following steps are taken:</p>

```

Comment:
this entire method is just "concat without IsConcatSpreadable", right?

is there a way we could make both methods mostly share the same AO?

Reply:
It also fills in holes too. Which could be a parameter of the AO I guess?

Reply:
Starts to get more complicated, but worth exploring.